sdk/eventhub/Azure.Messaging.EventHubs/design/event-processor{T}-proposal.md

```diff
@@ -0,0 +1,520 @@
+        int eventBatchSize,
+        string consumerGroup, 
+        string connectionString, 
+        EventProcessorOptions options = default);
+        
+    protected EventProcessor(
+        int eventBatchSize,
+        string consumerGroup, 
+        string connectionString, 
+        string eventHubName, 
+        EventProcessorOptions options = default);
+    
+    protected EventProcessor(
+        int eventBatchSize,
+        string consumerGroup,  
+        string fullyQualifiedNamespace, 
+        string eventHubName, 
+        TokenCredential credential, 
+        EventProcessorOptions options = default);
+    
+    // Optional extension points (core)
+    public virtual Task StartProcessingAsync(CancellationToken cancellationToken = default);
+    public virtual void StartProcessing(CancellationToken cancellationToken = default);
+    public virtual Task StopProcessingAsync(CancellationToken cancellationToken = default);
+    public virtual void StopProcessing(CancellationToken cancellationToken = default);
+    
+    protected virtual Task OnInitializingPartitionAsync(TPartition partition, CancellationToken cancellationToken);
+    protected virtual Task OnPartitionProcessingStoppedAsync(TPartition partition, ProcessingStoppedReason reason, CancellationToken cancellationToken);
+    
+    // Required extension points (core)
+    protected abstract Task OnProcessingEventBatchAsync(IEnumerable<EventData> events, TPartition partition, CancellationToken cancellationToken);
+    protected abstract Task OnProcessingErrorAsync(Exception exception, TPartition partition, string operationDescription, CancellationToken cancellationToken);
+    
+    // Required extension points (storage operations)
+    protected abstract Task<IEnumerable<EventProcessorCheckpoint>> ListCheckpointsAsync(CancellationToken cancellationToken);
+    protected abstract Task<IEnumerable<PartitionOwnership>> ListOwnershipAsync(CancellationToken cancellationToken);
+    protected abstract Task<IEnumerable<PartitionOwnership>> ClaimOwnershipAsync(IEnumerable<PartitionOwnership> desiredOwnership, CancellationToken cancellationToken);
+    
+    // Infrastructure
+    protected virtual LastEnqueuedEventProperties ReadLastEnqueuedEventProperties(string partitionId);
+}
+
+public class EventProcessorOptions 
+{
+    public EventHubConnectionOptions ConnectionOptions { get; set; }
+    public EventHubsRetryOptions RetryOptions { get; set; }
+    public string Identifier { get; set; }
+    public int? PrefetchCount { get; set; } 
+    public bool TrackLastEnqueuedEventProperties { get; set; }
+    public TimeSpan MaximumWaitTime { get; set; } = TimeSpan.FromSeconds(60);
+    public EventPosition DefaultStartingPosition { get; set; } = EventPosition.Earliest;
+    public TimeSpan LoadBalancingUpdateInterval { get; set; } = TimeSpan.FromSeconds(10);
+    public TimeSpan PartitionOwnershipExpirationInterval { get; set; } = TimeSpan.FromSeconds(30);
```

Comment:
Do you think we could have these in the `EventProcessorClient` as well?

Reply:
We probably should when we migrate over to deriving form this base.